main.tf

```diff
@@ -88,15 +92,18 @@ module "distribution_label" {
   tags       = "${var.tags}"
 }
 
-resource "null_resource" "default" {
-  triggers {
-    bucket             = "${element(compact(concat(list(var.origin_bucket), aws_s3_bucket.origin.*.bucket)), 0)}"
-    bucket_domain_name = "${format(var.bucket_domain_format, element(compact(concat(list(var.origin_bucket), aws_s3_bucket.origin.*.bucket)), 0))}"
-  }
+# ec2-downloads-windows is a bucket owned by amazon that will perminantly exist
```

Comment:
hmm...
not sure i like it for a few reasons:
1. The `ec2-downloads-windows` bucket can exist now, but can be changed later by AWS
2. Probably not a good idea to introduce this dependency - we'll have to maintain it
3. It's related to Windows (not a big deal, we just work with Linux :)
4. Data sources can introduce issue like race conditions etc.
5. From the code below, looks like that all we need from `aws_s3_bucket.selected` is the region. I suggest we provide the region in `var.region` as we do for many other modules. And remove `data "aws_region" "current" {}`. In this case, this module will not make any decisions on its own about regions and other data sources. That will be done in top-level modules, and we know the region we want to deploy to.

@osterman  what do you think about this?

Reply:
I think I'll be able to chime in once I understand the primary driver for using it.

Reply:
1. The region has to match where the bucket is. Or it doesn't work.
2. During destruction if `var.bucket` is `""` and the s3 bucket that would otherwise be created has already been destroyed. But for some reason terraform hits any other error during the destruction process. Then `local.bucket` can be left being `""`. This causes a following `terraform destroy` to fail because the `data.aws_s3_bucket` doesn't have a bucket to provide a region for.
While testing the module for each variable, i `applied` and `destroyed` tens of times, sometimes I got errors which I fixed. And thats how this workaround came to be.
If there is no bucket, then temporarily use a permanent aws public bucket during the destruction step.